### ISSUE TITLE
可以一份微调自定义数据集+grongding数据集的配置文件吗？

### ISSUE THREAD

**Issue by qiiiiiiiiiiiiiiiii** [AILab-CVC/YOLO-World#507]
我用作者给的微调方法成功的微调了自己的数据集，在封闭集上检测性能可以。但是泛化到其他类似的数据集效果就不行，同时我也试验了在一个数据集上微调6个类，测试8个类。同样效果不行。由此可以断定模型失去的zero-shot的能力。看到有人微调时加上grongding数据集，所以我想尝试一下。但是不知道该如何修改配置文件，希望作者和看到这个问题的人可以给一份相关的配置文件。

**Comment by qiiiiiiiiiiiiiiiii** [comment]
其实我也用作者提供的pretain的配置进行了修改，但有一些小问题，当我把num_training_classes设置为12时（自定义数据集的类别个数），就会报错。但我设置为80时，就可以跑通。
关于这点我还有疑问，作者pretain的数据集是ojb365+grongding数据集，那为什么num_training_classes是80，这里好像也没用到coco数据集

**Comment by qiiiiiiiiiiiiiiiii** [comment]
我目前是基于该yolo_world_v2_l_vlpan_bn_2e-3_100e_4x8gpus_obj365v1_goldg_train_1280ft_lvis_minival配置文件修改的，类别设置为80才能跑通，我想设置我自定义数据集的类别数目12。请问该怎么修改？


_base_ = ('../../third_party/mmyolo/configs/yolov8/'
          'yolov8_l_syncbn_fast_8xb16-500e_coco.py')
custom_imports = dict(imports=['yolo_world'],
                      allow_failed_imports=False)

# hyper-parameters
##修改
num_classes = 12
num_training_classes = 80

max_epochs = 20  # Maximum training epochs
close_mosaic_epochs = 2
save_epoch_intervals = 1
text_channels = 512
neck_embed_channels = [128, 256, _base_.last_stage_out_channels // 2]
neck_num_heads = [4, 8, _base_.last_stage_out_channels // 2 // 32]
base_lr = 2e-4
weight_decay = 0.025
#修改
train_batch_size_per_gpu = 2
load_from = "weights/yolo_world_v2_l_obj365v1_goldg_pretrain-a82b1fe3.pth"
# text_model_name = '../pretrained_models/clip-vit-base-patch32-projection'
text_model_name = 'openai/clip-vit-base-patch32'
img_scale = (1280, 1280)
classes=["Baton","Pliers","Hammer","Powerbank","Scissors","Wrench","Gun","Bullet","Sprayer","HandCuffs","Knife","Lighter"]
# model settings
model = dict(
    type='YOLOWorldDetector',
    mm_neck=True,
    num_train_classes=num_training_classes,
    num_test_classes=num_classes,
    data_preprocessor=dict(type='YOLOWDetDataPreprocessor'),
    backbone=dict(
        _delete_=True,
        type='MultiModalYOLOBackbone',
        image_model={{_base_.model.backbone}},
        text_model=dict(
            type='HuggingCLIPLanguageBackbone',
            model_name=text_model_name,
            frozen_modules=['all'])),
    neck=dict(type='YOLOWorldPAFPN',
              guide_channels=text_channels,
              embed_channels=neck_embed_channels,
              num_heads=neck_num_heads,
              block_cfg=dict(type='MaxSigmoidCSPLayerWithTwoConv')),
    bbox_head=dict(type='YOLOWorldHead',
                   head_module=dict(type='YOLOWorldHeadModule',
                                    use_bn_head=True,
                                    embed_dims=text_channels,
                                    num_classes=num_training_classes)),
    train_cfg=dict(assigner=dict(num_classes=num_training_classes)))

# dataset settings
text_transform = [
    dict(type='RandomLoadText',
         num_neg_samples=(num_classes, num_classes),
         max_num_samples=num_training_classes,
         padding_to_max=True,
         padding_value=''),
    dict(type='mmdet.PackDetInputs',
         meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape', 'flip',
                    'flip_direction', 'texts'))
]
train_pipeline = [
    *_base_.pre_transform,
    dict(type='MultiModalMosaic',
         img_scale=img_scale,
         pad_val=114.0,
         pre_transform=_base_.pre_transform),
    dict(
        type='YOLOv5RandomAffine',
        max_rotate_degree=0.0,
        max_shear_degree=0.0,
        scaling_ratio_range=(1 - _base_.affine_scale, 1 + _base_.affine_scale),
        max_aspect_ratio=_base_.max_aspect_ratio,
        border=(-img_scale[0] // 2, -img_scale[1] // 2),
        border_val=(114, 114, 114)),
    *_base_.last_transform[:-1],
    *text_transform,
]

train_pipeline_stage2 = [
    *_base_.pre_transform,
    dict(type='YOLOv5KeepRatioResize', scale=img_scale),
    dict(
        type='LetterResize',
        scale=img_scale,
        allow_scale_up=True,
        pad_val=dict(img=114.0)),
    dict(
        type='YOLOv5RandomAffine',
        max_rotate_degree=0.0,
        max_shear_degree=0.0,
        scaling_ratio_range=(1 - _base_.affine_scale, 1 + _base_.affine_scale),
        max_aspect_ratio=_base_.max_aspect_ratio,
        border_val=(114, 114, 114)),
    *_base_.last_transform[:-1],
    *text_transform
]

# obj365v1_train_dataset = dict(
#     type='MultiModalDataset',
#     dataset=dict(
#         type='YOLOv5Objects365V1Dataset',
#         data_root='data/PIDray/',
#         metainfo=dict(classes=classes),
#         ann_file='annotations/train.json',
#         data_prefix=dict(img='train/'),
#         filter_cfg=dict(filter_empty_gt=False, min_size=32)),
#     class_text_path='data/texts/PIDray.json',
#     pipeline=train_pipeline)

coco_train_dataset = dict(#_delete_=True,
                          type='MultiModalDataset',
                          dataset=dict(
                              type='YOLOv5CocoDataset',
#修改
                              data_root='data/PIDray',
                              ann_file='annotations/train.json',
                              metainfo=dict(classes=classes),
                              data_prefix=dict(img='train/'),

                              filter_cfg=dict(filter_empty_gt=False,
                                              min_size=32)),
#修改
                          class_text_path='data/texts/PIDray.json',
                          pipeline=train_pipeline)


mg_train_dataset = dict(type='YOLOv5MixedGroundingDataset',
                        data_root='data/mixed_grounding/',
                        ann_file='annotations/final_mixed_train_no_coco.json',
                        #修改
                        data_prefix=dict(img='images/'),
                        filter_cfg=dict(filter_empty_gt=False, min_size=32),
                        pipeline=train_pipeline)

flickr_train_dataset = dict(
    type='YOLOv5MixedGroundingDataset',
    data_root='data/flickr/',
    ann_file='annotations/final_flickr_separateGT_train.json',
    #修改
    data_prefix=dict(img='flickr30k-images/'),
    filter_cfg=dict(filter_empty_gt=True, min_size=32),
    pipeline=train_pipeline)

train_dataloader = dict(batch_size=train_batch_size_per_gpu,
                        collate_fn=dict(type='yolow_collate'),
                        dataset=dict(_delete_=True,
                                     type='ConcatDataset',
                                     datasets=[
                                         coco_train_dataset,
                                         flickr_train_dataset, mg_train_dataset
                                     ],
                                     ignore_keys=['classes', 'palette']))

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='YOLOv5KeepRatioResize', scale=img_scale),
    dict(
        type='LetterResize',
        scale=img_scale,
        allow_scale_up=False,
        pad_val=dict(img=114)),
    dict(type='LoadAnnotations', with_bbox=True, _scope_='mmdet'),
    dict(type='LoadText'),
    dict(type='mmdet.PackDetInputs',
         meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                    'scale_factor', 'pad_param', 'texts'))
]

coco_val_dataset = dict(
    _delete_=True,
    type='MultiModalDataset',
    dataset=dict(type='YOLOv5CocoDataset',
                 data_root='data/PIDray',
                 test_mode=True,
                 ann_file='annotations/test.json',
                 data_prefix=dict(img='test/'),
                 metainfo=dict(classes=classes),
                 batch_shapes_cfg=None),
    class_text_path='data/texts/PIDray.json',
    pipeline=test_pipeline)

val_dataloader = dict(dataset=coco_val_dataset)
test_dataloader = val_dataloader

val_evaluator = dict(type='mmdet.CocoMetric',
                     proposal_nums=(100, 1, 10),
                     classwise = True,
                     ann_file='data/PIDray/annotations/test.json',
                     metric='bbox')
test_evaluator = val_evaluator


# coco_val_dataset = dict(
#     _delete_=True,
#     type='MultiModalDataset',
#     dataset=dict(type='YOLOv5CocoDataset',
# #修改
#                  data_root='data/PIDray',
#                  ann_file='annotations/test.json',
#                  data_prefix=dict(img='test/'),
#                  metainfo=dict(classes=classes),
                 
#                  filter_cfg=dict(filter_empty_gt=False, min_size=32)),
# #修改
#     class_text_path='data/texts/PIDray.json',
#     pipeline=test_pipeline)
# val_dataloader = dict(dataset=coco_val_dataset)
# test_dataloader = val_dataloader
# # evaluation settings
# val_evaluator = dict(_delete_=True,
#                      type='mmdet.CocoMetric',
#                      proposal_nums=(100, 1, 10),
# #修改
#                      ann_file='data/PIDray/annotations/test.json',
#                      metric='bbox')


# training settings
default_hooks = dict(param_scheduler=dict(max_epochs=max_epochs),
                     checkpoint=dict(interval=save_epoch_intervals,
                                     rule='greater'))
custom_hooks = [
    dict(type='EMAHook',
         ema_type='ExpMomentumEMA',
         momentum=0.0001,
         update_buffers=True,
         strict_load=False,
         priority=49),
    dict(type='mmdet.PipelineSwitchHook',
         switch_epoch=max_epochs - close_mosaic_epochs,
         switch_pipeline=train_pipeline_stage2)
]
train_cfg = dict(max_epochs=max_epochs,
                 val_interval=10,
                 dynamic_intervals=[((max_epochs - close_mosaic_epochs),
                                     _base_.val_interval_stage2)])

optim_wrapper = dict(optimizer=dict(
    _delete_=True,
    type='AdamW',
    lr=base_lr,
    weight_decay=weight_decay,
    batch_size_per_gpu=train_batch_size_per_gpu),
                     paramwise_cfg=dict(bias_decay_mult=0.0,
                                        norm_decay_mult=0.0,
                                        custom_keys={
                                            'backbone.text_model':
                                            dict(lr_mult=0.01),
                                            'logit_scale':
                                            dict(weight_decay=0.0)
                                        }),
                     constructor='YOLOWv5OptimizerConstructor')

**Comment by yfllllll** [comment]
@qiiiiiiiiiiiiiiiii 你的疑问解决了吗？这个参数究竟是什么意思啊
